src/workflows-app/SubmissionConfig.test.js

```diff
@@ -285,3 +302,1391 @@ describe('BaseSubmissionConfig gets proxy urls from Leo', () => {
+          describeAllRecordTypes: mockTypesResponse,
+        },
+        WorkflowScript: {
+          get: mockWdlResponse,
+        },
+      };
+    });
+
+    // ** ACT **
+    await act(async () =>
+      render(
+        h(BaseSubmissionConfig, {
+          methodId: '123',
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      )
+    );
+
+    // ** ASSERT **
+    await waitFor(() => {
+      expect(mockRunSetResponse).toHaveBeenCalledTimes(1);
+      expect(mockTypesResponse).toHaveBeenCalledTimes(1);
+      expect(mockMethodsResponse).toHaveBeenCalledTimes(1);
+      expect(mockSearchResponse).toHaveBeenCalledTimes(1);
+      expect(mockWdlResponse).toHaveBeenCalledTimes(1);
+    });
+
+    const table = await screen.findByRole('table');
+
+    const rows = within(table).queryAllByRole('row');
+    expect(rows.length).toBe(5);
+
+    const headers = within(rows[0]).queryAllByRole('columnheader');
+    expect(headers.length).toBe(4);
+
+    const cells = within(rows[1]).queryAllByRole('cell');
+    expect(cells.length).toBe(4);
+  });
+
+  it('should initially populate the inputs definition table with attributes determined by the previously executed run set', async () => {
+    // ** ARRANGE **
+    const mockRunSetResponse = jest.fn(() => Promise.resolve(runSetResponse));
+    const mockMethodsResponse = jest.fn(() => Promise.resolve(methodsResponse));
+    const mockSearchResponse = jest.fn((_root, _instanceId, recordType) => Promise.resolve(searchResponses[recordType]));
+    const mockTypesResponse = jest.fn(() => Promise.resolve(typesResponse));
+    const mockWdlResponse = jest.fn(() => Promise.resolve('mock wdl response'));
+
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            getForMethod: mockRunSetResponse,
+          },
+          methods: {
+            getById: mockMethodsResponse,
+          },
+        },
+        WorkspaceData: {
+          queryRecords: mockSearchResponse,
+          describeAllRecordTypes: mockTypesResponse,
+        },
+        WorkflowScript: {
+          get: mockWdlResponse,
+        },
+      };
+    });
+
+    // ** ACT **
+    render(
+      h(BaseSubmissionConfig, {
+        methodId: '123',
+        name: 'test-azure-ws-name',
+        namespace: 'test-azure-ws-namespace',
+        workspace: mockAzureWorkspace,
+      })
+    );
+
+    // ** ASSERT **
+    await waitFor(() => {
+      expect(mockRunSetResponse).toHaveBeenCalledTimes(1);
+      expect(mockTypesResponse).toHaveBeenCalledTimes(1);
+      expect(mockMethodsResponse).toHaveBeenCalledTimes(1);
+      expect(mockSearchResponse).toHaveBeenCalledTimes(1);
+      expect(mockWdlResponse).toHaveBeenCalledTimes(1);
+    });
+
+    const button = await screen.findByRole('button', { name: 'Inputs' });
+
+    // ** ACT **
+    await userEvent.click(button);
```

Comment:
Testing Library recommends creating an instance of UserEvent using `userEvent.setup()` and calling methods on it rather than calling methods directly on the `userEvent` import.

https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent